### PR TITLE
Added a package for the MDAnalysis toolkit.

### PR DESCRIPTION
**Pull request by bvanessen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class PyMdanalysis(PythonPackage):
+    """MDAnalysis is a Python toolkit to analyze molecular dynamics
+    trajectories generated by a wide range of popular simulation
+    packages including DL_Poly, CHARMM, Amber, NAMD, LAMMPS, and
+    Gromacs. (See the lists of supported trajectory formats and
+    topology formats.)"""
+
+    homepage = "http://www.mdanalysis.org"
+    url      = "https://pypi.python.org/packages/9f/56/a4cae14e45aa893a3578905a97e17a325d663e424736366adbc134e72dce/MDAnalysis-0.15.0.tar.gz#md5=19e5a8e6c2bfe85f6209d1d7a36e4f20"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: Can you change this to:
https://pypi.io/packages/source/M/MDAnalysis/MDAnalysis-0.15.0.tar.gz
This will allow us to download different versions from the same URL.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    version('0.15.0', '19e5a8e6c2bfe85f6209d1d7a36e4f20')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: It looks like "MDAnalysis requires Python 2.7 or better". Can you add:
```python
depends_on('python@2.7:')
```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: It looks like the package uses cython if found to optimize the code. Can you add:
```python
depends_on('py-cython@0.16:', type='build')
```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: From the `setup.py`: "MDAnalysis requires a version of NumPy (>=1.5.0), even for setup". Please add:
```python
depends_on('py-numpy@1.5.0:', type=('build', 'run'))
```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: This package is in fact missing several other dependencies as well:
```python
install_requires=[                                                    
    'numpy>=1.5.0',                                                   
    'biopython>=1.59',                                                
    'networkx>=1.0',                                                  
    'GridDataFormats>=0.3.2',                                         
    'six>=1.4.0',                                                     
],
```
and optionally:
```python
extras_require={                                                      
    'AMBER': ['netCDF4>=1.0'],  # for AMBER netcdf, also needs HDF5   
                                # and netcdf-4                        
    'analysis': [                                                     
        'matplotlib',                                                 
        'scipy',                                                      
        'seaborn',  # for annotated heat map and nearest neighbor     
                    # plotting in PSA                                 
    ],                                                                
},
```
You don't have to add the optional dependencies, but if you want the support you can add these as variants.

**Comment by bvanessen** [comment]
I tried using the link that you provided, but it doesn’t work with curl.  There is a redirect that will not process correctly.  I also tried several other versions that used variants of that link including,

url      = "https://pypi.io/packages/source/M/MDAnalysis/MDAnalysis-0.15.0.tar.gz"
url      = "https://pypi.python.org/packages/source/M/MDAnalysis/MDAnalysis-0.15.0.tar.gz"
url      = "https://files.pythonhosted.org/packages/source/M/MDAnalysis/MDAnalysis-0.15.0.tar.gz”


and none of these worked correctly.  They all had some sort of 404 or redirect message.


Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470

**Comment by adamjstewart** [comment]
There's already a `py-seaborn` package in Spack (I think I wrote it).

**Comment by bvanessen** [comment]
I saw that after I added it.  I checked my local repo and forgot to pull the most recent version of the repo.

File: var/spack/repos/builtin/packages/py-griddataformats/package.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    files, make them available as a Grid object, and write out the
+    data again."""
+
+    homepage = "http://www.mdanalysis.org/GridDataFormats"
+    url      = "https://pypi.io/packages/source/G/GridDataFormats/GridDataFormats-0.3.3.tar.gz"
+
+    version('0.3.3', '5c83d3bdd421eebcee10111942c5a21f')
+
+    depends_on('python@2.7:')
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-griddataformats/package.py]: This package is missing a couple dependencies:
```python
install_requires=['numpy>=1.0.3', 'six'],                                 
```
This can be found in the `setup.py`. I also don't see where it says that it doesn't work with older versions of Python.

File: var/spack/repos/builtin/packages/py-keras/package.py [hunk 4]
```diff
@@ -40,6 +40,6 @@ class PyKeras(PythonPackage):
     version('1.1.0', 'd1711362ac8473238b0d198d2e3a0574')
 
     depends_on('py-setuptools', type='build')
-    depends_on('py-theano', type=('build', 'run'))
+    depends_on('py-theano', type=('build'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keras/package.py]: Are you sure this is only needed at build time? In the tarball I see a lot of `import theano`.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    depends_on('py-six@1.9.0:', type=('build', 'run'))
+    depends_on('py-biopython@1.59:', type=('build', 'run'))
+    depends_on('py-networkx@1.0:', type=('build', 'run'))
+    depends_on('py-griddataformats@0.3.2:', type=('build', 'run'))
+
+    depends_on('py-matplotlib', when='+analysis', type=('build', 'run'))
+    depends_on('py-scipy', when='+analysis', type=('build', 'run'))
+    depends_on('py-seaborn', when='+analysis', type=('build', 'run'))
+
+    depends_on('py-netcdf@1.0:', when='+amber', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: This package was renamed to `py-netcdf4`. There is actually a different `py-netcdf` on PyPI, but it isn't in Spack.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    depends_on('hdf5', when='+amber', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: It doesn't need to be a link dependency?

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    variant('analysis', default=True, 
+            description='Enable analysis packages: matplotlib, scipy, seaborn')
+    variant('amber', default=False,
+            description='Support AMBER netcdf format.')
+
+    depends_on('python@2.7:')
+    depends_on('py-setuptools', type='build')
+    depends_on('py-cython@0.16:', type='build')
+    depends_on('py-numpy@1.5.0:', type=('build', 'run'))
+    depends_on('py-six@1.9.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: `@1.4.0:`

File: var/spack/repos/builtin/packages/lbann/package.py [hunk 8]
```diff
@@ -46,7 +46,7 @@ class Lbann(CMakePackage):
     depends_on('cuda', when='+gpu')
     depends_on('mpi')
     depends_on('opencv@3.2.0', when='+opencv')
-    depends_on('protobuf@3.0.2')
+    depends_on('protobuf@3.2.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lbann/package.py]: Why the change?

File: var/spack/repos/builtin/packages/elemental/package.py [hunk 9]
```diff
@@ -67,9 +67,18 @@ class Elemental(CMakePackage):
...
-    depends_on('openblas +openmp', when='+openmp_blas ~int64_blas')
+#    depends_on('openblas +openmp', when='+openmp_blas ~int64_blas')
+#    depends_on('openblas +openmp', when='+openmp_blas ~int64_blas %gcc')
+#    depends_on('openblas +openmp', when='+openmp_blas ~int64_blas %clang')
+#    depends_on('openblas +openmp', when='+openmp_blas ~int64_blas %xl')
+#    depends_on('openblas +openmp', when='+openmp_blas ~int64_blas %intel')
+#    depends_on('intel-mkl', when='%intel@17.0.2:')
+    depends_on('intel-mkl', when='+openmp_blas %intel@17.0.2:')
+    depends_on('intel-mkl +openmp', when='+openmp_blas ~int64_blas %intel@17.0.2:')
+    depends_on('intel-mkl +openmp +ilp64', when='+openmp_blas +int64_blas %intel@17.0.2:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/elemental/package.py]: Can you explain these changes a little better?
